WooCommerce/Classes/ViewRelated/Products/Edit Product/Linked Products List Selector/LinkedProductListSelectorDataSource.swift

```diff
@@ -123,3 +123,19 @@ extension LinkedProductListSelectorDataSource {
         return linkedProductIDs != originalLinkedProductIDs
     }
 }
+
+extension LinkedProductListSelectorDataSource: DraggablePaginatedListSelectorDataSource {
+
+    /// Called when the user rearranges products.
+    func moveItem(from sourceIndex: Int, to destinationIndex: Int) {
+        guard sourceIndex < linkedProductIDs.count else {
```

Comment:
nit: maybe we can also early return when `sourceIndex` is the same as `destinationIndex`? just a minor optimization